pkg/restic/common.go

```diff
@@ -125,6 +135,47 @@ func GetVolumesToBackup(obj metav1.Object) []string {
 	return strings.Split(backupsValue, ",")
 }
 
+func getVolumesToExclude(obj metav1.Object) []string {
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		return nil
+	}
+
+	return strings.Split(annotations[VolumesToExcludeAnnotation], ",")
+}
+
+func contains(list []string, k string) bool {
+	for _, i := range list {
+		if i == k {
+			return true
+		}
+	}
+	return false
+}
+
+// GetPodVolumesUsingRestic returns a list of volume names to backup for the provided pod.
+func GetPodVolumesUsingRestic(pod *corev1api.Pod, defaultRestic bool) []string {
+	if !defaultRestic {
+		return GetVolumesToBackup(pod)
+	}
+
+	volsToExclude := getVolumesToExclude(pod)
+	podVolumes := []string{}
+	for _, pv := range pod.Spec.Volumes {
+		// cannot backup hostpath volumes as they are not mounted into /var/lib/kubelet/pods
+		// and therefore not accessible to the restic daemon set.
+		if pv.HostPath != nil {
+			continue
+		}
+		// don't include volumes that are in the exclude list
+		// and the volume mounting the default service account token.
+		if !contains(volsToExclude, pv.Name) && !strings.HasPrefix(pv.Name, "default-token") {
```

Comment:
I would probably invert this logic to remain consistent with the previous host path case (and I think this is considered more idiomatic go), i.e.:

```go
if contains(volsToExclude, pv.Name) {
    continue
}
if strings.HasPrefix(pv.Name, "default-token") {
    continue
}

podVolumes = append(podVolumes, pv.Name)
```

Reply:
done